src/api_routes_v4.rs

```diff
@@ -0,0 +1,392 @@
+  tagline::{
+    create::create_tagline,
+    delete::delete_tagline,
+    list::list_taglines,
+    update::update_tagline,
+  },
+  user::{
+    create::{authenticate_with_oauth, register},
+    delete::delete_account,
+    my_user::get_my_user,
+  },
+};
+use lemmy_apub::api::{
+  list_comments::list_comments,
+  list_posts::list_posts,
+  read_community::get_community,
+  read_person::read_person,
+  resolve_object::resolve_object,
+  search::search,
+  user_settings_backup::{export_settings, import_settings},
+};
+use lemmy_routes::images::image_proxy;
+use lemmy_utils::rate_limit::RateLimitCell;
+
+pub fn config(cfg: &mut ServiceConfig, rate_limit: &RateLimitCell) {
+  cfg.service(
+    scope("/api/v4")
+      .wrap(rate_limit.message())
+      .route("/image_proxy", get().to(image_proxy))
+      // Site
+      .service(
+        scope("/site")
+          .route("", get().to(get_site_v4))
+          .route("", post().to(create_site))
+          .route("", put().to(update_site)),
+      )
+      .route("/modlog", get().to(get_mod_log))
+      .service(
+        resource("/search")
+          .wrap(rate_limit.search())
+          .route(get().to(search)),
+      )
+      .route("/resolve_object", get().to(resolve_object))
+      // Community
+      .service(
+        resource("/community")
+          .guard(guard::Post())
+          .wrap(rate_limit.register())
+          .route(post().to(create_community)),
+      )
+      .service(
+        scope("/community")
+          .route("", get().to(get_community))
+          .route("", put().to(update_community))
+          .route("/random", get().to(get_random_community))
+          .route("/hide", put().to(hide_community))
+          .route("/list", get().to(list_communities))
+          .route("/follow", post().to(follow_community))
+          .route("/delete", post().to(delete_community))
+          // Mod Actions
+          .route("/remove", post().to(remove_community))
+          .route("/transfer", post().to(transfer_community))
+          .route("/ban_user", post().to(ban_from_community))
+          .route("/mod", post().to(add_mod_to_community))
+          .service(
+            scope("/pending_follows")
+              .route("/count", get().to(get_pending_follows_count))
+              .route("/list", get().to(get_pending_follows_list))
+              .route("/approve", post().to(post_pending_follows_approve)),
+          ),
+      )
+      .route("/federated_instances", get().to(get_federated_instances))
+      // Post
+      .service(
+        // Handle POST to /post separately to add the post() rate limitter
+        resource("/post")
+          .guard(guard::Post())
+          .wrap(rate_limit.post())
+          .route(post().to(create_post)),
+      )
+      .service(
+        scope("/post")
+          .route("", get().to(get_post))
+          .route("", put().to(update_post))
+          .route("/delete", post().to(delete_post))
+          .route("/remove", post().to(remove_post))
```

Comment:
Is there any idea for better distinguishing what's currently called "delete" and "remove"? Changing it later would be less convenient.

Reply:
I think delete (for content you created) vs remove (for content others created) is probably fine. 

I thought about whether we should move all of the mod related actions to a mod subheading, but its probably better to keep them at the item level like they are now (`/comment/remove`), or else we'd just have to replicate the same things but under a different heading (`/mod/comment/remove`)